### ISSUE TITLE
User can be registered  via AmazonCognitoYourUserPoolsDemo, but cannot be signed in

### ISSUE THREAD

**Issue by guipingzhang** [aws/aws-sdk-android#146]
I can successfully sign up a user to User Pool and confirm the signup. In User Pool console, I can find this user. 

However, when I sign in by using correct Username and Password, it shows the message: Error: com.amazonaws.services.cognitoidentityprovider.modle.NotAuthorizedException: Incorrect user
sername or password.   (Service: AmazonCognitoIdentityProvider;Status Code: 400; Error Code: NotAuthorizedException; Request ID: 29544def-291d-11e6-bc73-e9fff913235b)

Is there any suggestion to this kind of error?

**Comment by chetanme** [comment]
Are you using the version 2.2.17 or higher? We had locale related bug in the SDKs which is now fixed.

If you are still getting the same exception can you post the time stamp for this request id?

**Comment by guipingzhang** [comment]
Thanks for your reply. I downloaded aws-sdk-android-release_v2.2.17 and included the dependency in build.gradle(Module:app) as follows,
dependencies {
    compile 'com.android.support:appcompat-v7:23.2.1'
    compile 'com.android.support:design:23.1.1'
    compile 'com.amazonaws:aws-android-sdk-cognitoidentityprovider:2.2.17+'
}I registered a new user (username=huang) in my user pool (poolname=GP) and still failed to sign in. 06-03 14:47:42.889 25412-25412/com.amazonaws.youruserpools.CognitoYourUserPoolsDemo E/App Error: com.amazonaws.services.cognitoidentityprovider.model.NotAuthorizedException: Incorrect username or password. (Service: AmazonCognitoIdentityProvider; Status Code: 400; Error Code: NotAuthorizedException; Request ID: 0b30cd28-2957-11e6-84ce-bbbb4c899843)
Is there any further suggestion? Thanks a lot.Date: Thu, 2 Jun 2016 21:48:07 -0700

**Comment by chetanme** [comment]
I checked the logs and the failure was due to an incorrect password being passed from the client. If you are sure that you are using the correct username password combination then it could be a device issue. Are you doing this on an old emulator? Can you try doing this on a different device?

**Comment by guipingzhang** [comment]
The device is HUAWEISCL-TLOOH android 5.1.1 API 22. 
06-04 10:16:17.195 18906-18906/com.amazonaws.youruserpools.CognitoYourUserPoolsDemo E/RegisterUser username:: abcdef
06-04 10:16:17.195 18906-18906/com.amazonaws.youruserpools.CognitoYourUserPoolsDemo E/RegisterUser password:: abcdef

06-04 10:53:35.355 17122-17122/com.amazonaws.youruserpools.CognitoYourUserPoolsDemo E/MainActivity username:: abcdef
06-04 10:53:35.355 17122-17122/com.amazonaws.youruserpools.CognitoYourUserPoolsDemo E/MainActivity password:: abcdef
06-04 10:53:35.355 17122-17122/com.amazonaws.youruserpools.CognitoYourUserPoolsDemo E/Authen return user:: abcdef
06-04 10:53:35.355 17122-17122/com.amazonaws.youruserpools.CognitoYourUserPoolsDemo E/Authen return pass:: abcdef

06-04 10:53:39.065 17122-17122/com.amazonaws.youruserpools.CognitoYourUserPoolsDemo E/App Error: com.amazonaws.services.cognitoidentityprovider.model.NotAuthorizedException: Incorrect username or password. (Service: AmazonCognitoIdentityProvider; Status Code: 400; Error Code: NotAuthorizedException; Request ID: 8941d132-29ff-11e6-9dac-43d507a34bc9)


I failed to launch AVD emulator,

**Comment by mreddyg** [comment]
@guipingzhang 
We have been unable to reproduce this issue in our testing. Would you be willing to help us debug this problem? I can create a new sample app with some debug statements to help us understand whats happening during the authentication process. The debug statements will add checkpoints and data related to SRP calculations to logcat - which you too will be able to see in the logs.

If you agree to do this, I will create the sample have our customer reach out team to deliver it to you. The sample and the logs will be delivered through them, you will not have to post them on public forums.

Thanks,
Mahesh

**Comment by guipingzhang** [comment]
@mreddyg Dear Mahesh, thank you so much. I am very pleased to do so, as I highly appreciate the convenience of User Pool for user register and login. Presently, I use the sample located at https://github.com/awslabs/aws-sdk-android-samples/AmazonCognitoYourUserPoolsDemo.

**Comment by mreddyg** [comment]
@guipingzhang, Thank you for agreeing to help us! Our customer reach out team will contact you in a couple of days with a new sample app. This will be same as the sample in the GitHub, but will contain logging in the SDK. After installing the new sample app on your phone, you will have to collect logs from logcat when you reproduce the issue and send us the logs through the customer reach out team.

**Comment by UnCheon** [comment]
I think your Id was not verified. If you verify your username by phone number or  email, you can sign in without the error.

**Comment by guipingzhang** [comment]
@UnCheon  Thank you for your reply. In early June, I had verified Id via the verification code sent to my email and couldn't log in (previously I set the policies: minimum length of password is  6 and all lowercase letters). 

Today, I create a new User Pool. I can successfully register a new user and can login with this user. I follow the default policy ( minimum length of password is 8 and with number, uppercase letters, lowercase letters and special character). 

Thank you and your great group.
